src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -304,29 +316,24 @@ private static object GetWebHookData(Type dataType, WebHookHandlerContext contex
             return getDataMethod.Invoke(null, new object[] { context });
         }
 
-        public FunctionDescriptor GetHttpFunctionOrNull(Uri uri)
+        public FunctionDescriptor GetHttpFunctionOrNull(HttpRequestMessage request)
         {
-            if (uri == null)
+            if (request == null)
             {
-                throw new ArgumentNullException("uri");
+                throw new ArgumentNullException(nameof(request));
             }
 
             if (HttpFunctions == null || HttpFunctions.Count == 0)
             {
                 return null;
             }
 
-            // Parse the route (e.g. "api/myfunc") to get 'myfunc"
-            // including any path after "api/"
             FunctionDescriptor function = null;
-            string route = HttpUtility.UrlDecode(uri.AbsolutePath);
-            int idx = route.ToLowerInvariant().IndexOf("api", StringComparison.OrdinalIgnoreCase);
-            if (idx > 0)
+            var routeData = HttpRoutes.GetRouteData(request);
```

Comment:
The behavior is (I just wrote a test for it) is that the first match is used. No error is thrown.